docs: add doc for inference (#514)
Azure/kaito: docs/inference/README.md

# Kaito Inference Workspace API

> This guide provides instructions on how to use the Kaito Inference Workspace API for parameter-efficient fine-tuning (PEFT) of models.

Reviewer: ```suggestion
This guide provides instructions on how to use the Kaito Inference Workspace API for basic model serving and serving with LoRA adapters..
```

Example 1: Inferencing [`phi-3-mini`](../../examples/inference/kaito_workspace_phi_3.yaml)

Example 2: Inferencing [`falcon-7b`](../../examples/inference/kaito_workspace_falcon_7b.yaml) without adapters

Example 3: Inferencing `falcon-7b` with adapters

```yaml
apiVersion: kaito.sh/v1alpha1
kind: Workspace
metadata:
  name: workspace-falcon-7b
resource:
  instanceType: "Standard_NC12s_v3"
  labelSelector:
    matchLabels:
      apps: falcon-7b
inference:
  preset:
    name: "falcon-7b"
  adapters:
    - source:
        name: "falcon-7b-adapter"
        image:  "<YOUR_IMAGE>"
      strength: "0.2"
```

Multiple adapters can be added:

```yaml
apiVersion: kaito.sh/v1alpha1
kind: Workspace
metadata:
  name: workspace-falcon-7b
resource:
  instanceType: "Standard_NC12s_v3"
  labelSelector:
    matchLabels:
      apps: falcon-7b
inference:
  preset:
    name: "falcon-7b"
  adapters:
    - source:
        name: "falcon-7b-adapter"
        image:  "<YOUR_IMAGE>"
      strength: "0.2"
    - source:
        name: "additional-source"
        image: "<YOUR_ADDITIONAL_IMAGE>"
      strength: "0.5" 
```

> Only images are supported as adapters source for now, and the default strength of adapters will be "1.0". Only images are supported as adapters source for now.

Reviewer 2: ```suggestion
Currently, only images are supported as adapter sources, with a default strength of "1.0".
```
